polkadot/xcm/src/lib.rs

```diff
@@ -258,52 +414,68 @@ versioned_type! {
 	pub enum VersionedJunction {
 		#[codec(index = 2)]
 		V2(v2::Junction),
-		#[codec(index = 3)]
+		#[codec(index = 3)] // Same as v4
 		V3(v3::Junction),
 	}
 }
 
 versioned_type! {
-	/// A single `MultiLocation` value, together with its version code.
+	/// A single `Location` value, together with its version code.
 	#[derive(Ord, PartialOrd)]
-	pub enum VersionedMultiLocation {
+	pub enum VersionedLocation {
 		#[codec(index = 1)] // v2 is same as v1 and therefore re-using the v1 index
 		V2(v2::MultiLocation),
 		#[codec(index = 3)]
 		V3(v3::MultiLocation),
+		#[codec(index = 4)]
+		V4(v4::Location),
 	}
 }
 
+pub type VersionedMultiLocation = VersionedLocation;
+
 versioned_type! {
-	/// A single `InteriorMultiLocation` value, together with its version code.
-	pub enum VersionedInteriorMultiLocation {
+	/// A single `InteriorLocation` value, together with its version code.
+	pub enum VersionedInteriorLocation {
 		#[codec(index = 2)] // while this is same as v1::Junctions, VersionedInteriorMultiLocation is introduced in v3
 		V2(v2::InteriorMultiLocation),
 		#[codec(index = 3)]
 		V3(v3::InteriorMultiLocation),
+		#[codec(index = 4)]
+		V4(v4::InteriorLocation),
 	}
 }
 
+pub type VersionedInteriorMultiLocation = VersionedInteriorLocation;
+
 versioned_type! {
-	/// A single `MultiAsset` value, together with its version code.
-	pub enum VersionedMultiAsset {
+	/// A single `Asset` value, together with its version code.
+	pub enum VersionedAsset {
 		#[codec(index = 1)] // v2 is same as v1 and therefore re-using the v1 index
 		V2(v2::MultiAsset),
 		#[codec(index = 3)]
 		V3(v3::MultiAsset),
+		#[codec(index = 4)]
+		V4(v4::Asset),
 	}
 }
 
+pub type VersionedMultiAsset = VersionedAsset;
+
 versioned_type! {
 	/// A single `MultiAssets` value, together with its version code.
-	pub enum VersionedMultiAssets {
+	pub enum VersionedAssets {
 		#[codec(index = 1)] // v2 is same as v1 and therefore re-using the v1 index
 		V2(v2::MultiAssets),
 		#[codec(index = 3)]
 		V3(v3::MultiAssets),
+		#[codec(index = 4)]
+		V4(v4::Assets),
 	}
 }
 
+pub type VersionedMultiAssets = VersionedAssets;
```

Comment:
Might also want to do this:

```suggestion
#[deprecated(since = "4", note = "Use `VersionedAssets` instead")]
pub type VersionedMultiAssets = VersionedAssets;
```